rosbridge_server/scripts/rosbridge_websocket.py

```diff
@@ -55,184 +60,240 @@
-    RosbridgeWebSocket.client_count_pub.publish(0)
-
-    # Get the glob strings and parse them as arrays.
-    RosbridgeWebSocket.topics_glob = [
-        element.strip().strip("'")
-        for element in rospy.get_param('~topics_glob', '')[1:-1].split(',')
-        if len(element.strip().strip("'")) > 0]
-    RosbridgeWebSocket.services_glob = [
-        element.strip().strip("'")
-        for element in rospy.get_param('~services_glob', '')[1:-1].split(',')
-        if len(element.strip().strip("'")) > 0]
-    RosbridgeWebSocket.params_glob = [
-        element.strip().strip("'")
-        for element in rospy.get_param('~params_glob', '')[1:-1].split(',')
-        if len(element.strip().strip("'")) > 0]
-
-    if "--port" in sys.argv:
-        idx = sys.argv.index("--port")+1
-        if idx < len(sys.argv):
-            port = int(sys.argv[idx])
-        else:
-            print("--port argument provided without a value.")
-            sys.exit(-1)
-
-    if "--address" in sys.argv:
-        idx = sys.argv.index("--address")+1
-        if idx < len(sys.argv):
-            address = int(sys.argv[idx])
-        else:
-            print("--address argument provided without a value.")
-            sys.exit(-1)
-
-    if "--retry_startup_delay" in sys.argv:
-        idx = sys.argv.index("--retry_startup_delay") + 1
-        if idx < len(sys.argv):
-            retry_startup_delay = int(sys.argv[idx])
-        else:
-            print("--retry_startup_delay argument provided without a value.")
-            sys.exit(-1)
-
-    if "--fragment_timeout" in sys.argv:
-        idx = sys.argv.index("--fragment_timeout") + 1
-        if idx < len(sys.argv):
-            RosbridgeWebSocket.fragment_timeout = int(sys.argv[idx])
-        else:
-            print("--fragment_timeout argument provided without a value.")
-            sys.exit(-1)
-
-    if "--delay_between_messages" in sys.argv:
-        idx = sys.argv.index("--delay_between_messages") + 1
-        if idx < len(sys.argv):
-            RosbridgeWebSocket.delay_between_messages = float(sys.argv[idx])
-        else:
-            print("--delay_between_messages argument provided without a value.")
-            sys.exit(-1)
-
-    if "--max_message_size" in sys.argv:
-        idx = sys.argv.index("--max_message_size") + 1
-        if idx < len(sys.argv):
-            value = sys.argv[idx]
-            if value == "None":
-                RosbridgeWebSocket.max_message_size = None
+
+class RosbridgeWebsocketNode(Node):
+    def __init__(self):
+        super().__init__('rosbridge_websocket')
+
+        ##################################################
+        # Parameter handling                             #
+        ##################################################
+        retry_startup_delay = self.get_parameter('retry_startup_delay').value
+        retry_startup_delay = 2.0 if retry_startup_delay is None
+                                  else retry_startup_delay  # seconds
```

Comment:
I think you can use `get_parameter_or` to avoid this logic, right?

Reply:
same in many of the other calls to `get_parameter` here

Reply:
Yes; that wasn't part of `rclpy` back then when I made these changes.
+1 to changing it to `get_parameter_or`.

Reply:
Changed to `declare_parameter` in Thanks; addressed in https://github.com/RobotWebTools/rosbridge_suite/pull/419/commits/9161521da0b402b5fa6720e54cb61bd156ec58f6.